src/Controls/src/Core/Element/Element.cs

```diff
@@ -587,6 +587,8 @@ protected virtual void OnChildAdded(Element child)
 		{
 			child.SetParent(this);
 
+			child.ApplyBindings();
```

Comment:
This change was needed after this PR: https://github.com/dotnet/maui/pull/24553

Reply:
This is something I removed on purpose from my PR, may you explain why this is needed?

`SetParent` propagates the `BindingContext` which will automatically call `ApplyBindings`.

This change you put here will break a unit test.

Reply:
@kubaflo I tested your (reverted) UI test page and added this:

```
<ContentPage.ToolbarItems>
  <ToolbarItem>
    <ToolbarItem.IconImageSource>
      <FontImageSource Glyph="&#xf133;"
                       x:Name="TheIcon"
                       Color="{AppThemeBinding Light=Black, Dark=White}"
                       FontFamily="FA"/>
    </ToolbarItem.IconImageSource>
  </ToolbarItem>
</ContentPage.ToolbarItems>
<VerticalStackLayout>
  <Label Text="{Binding Color, Source={x:Reference TheIcon}}" />
```

You can clearly see that the color changes even without this `child.ApplyBindings()`, which means that the binding is working correctly and probably the real issue is somewhere else.

![Simulator Screen Recording - iPhone Xs - 2024-09-14 at 11 55 12](https://github.com/user-attachments/assets/6e27c3fd-160c-49b1-8c08-f776251565b6)

Reply:
Probably the best way to achieve what you want is to act on `PrimaryToolbarItem` and where `item.PropertyChanged += OnPropertyChanged;` is subscribed, also subscribe to `item.IconImageSource.PropertyChanged += OnIconPropertyChanged;`.. something like that.

Reply:
@albyrock87  This PR worked before this https://github.com/dotnet/maui/pull/24553 and your changes broke it😅 I'm not saying it is bad, but maybe something worth looking into... It's because if it broke a fix in this pr then maybe it breaks something more important that already exists in the codebase🤔

Reply:
That's exactly why I've investigated your report in depth, and now I'm completely sure that the issue is in the navigation renderer and not in Element.

That line on Element was basically applying the Bindings twice, which probably had a side effect of retriggering some method in the navigation renderer.

Reply:
I got you but, I've just checked and it has broken Android, which I didn't touch.

`Before`

https://github.com/user-attachments/assets/9d8914bd-c93f-4654-bb9b-63ac99748c4f

`After:`

https://github.com/user-attachments/assets/2686f90b-aae3-414d-9f93-aa3d083baec9

Reply:
We can probably fix them all from inside ToolbarItem by triggering property changed of IconImageSource when IconImageSource.SourceChanged.

Reply:
I confirm this fixes everything (`MenuItem.cs`).

```cs
/// <summary>Bindable property for <see cref="IconImageSource"/>.</summary>
public static readonly BindableProperty IconImageSourceProperty = BindableProperty.Create(nameof(IconImageSource), typeof(ImageSource), typeof(MenuItem), default(ImageSource),
	propertyChanged: (bindable, oldValue, newValue) => {
		if (oldValue is ImageSource oldSource)
		{
			oldSource.SourceChanged -= ((MenuItem)bindable).OnImageSourceSourceChanged;
		}

		((MenuItem)bindable).AddRemoveLogicalChildren(oldValue, newValue);
		
		if (newValue is ImageSource newSource)
		{
			newSource.SourceChanged += ((MenuItem)bindable).OnImageSourceSourceChanged;
		}
	}
);

private void OnImageSourceSourceChanged(object sender, EventArgs e)
{
	OnPropertyChanged(IconImageSourceProperty.PropertyName);
}
```

Now this is not a "clean" solution, but it avoids subscribing to `SourceChanged` on each platform.

Reply:
Yea, it works and this solution is the same as in `imageCell.cs` which leads me to believe that it is the right one!

https://github.com/kubaflo/maui/blob/0a6df3601695f0ca84e63dd70a06877a14197b6f/src/Controls/src/Core/Cells/ImageCell.cs#L9-L13